src/llm/feature_abstract_container_color.cc

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file feature_abstract_container_color.cc
+ *
+ * @brief
+ *
+ * Class used to describe a feature of the abstract_container color
+ */
+#include <sstream>
+
+#include "h2sl/feature_abstract_container_color.h"
+#include "h2sl/abstract_container.h"
+
+using namespace std;
+using namespace h2sl;
+
+/**
+ * Feature_Abstract_Container_Color class constructor
+ */
+Feature_Abstract_Container_Color::
+Feature_Abstract_Container_Color( const bool& invert,
+                          const std::string& abstract_container_color ) : Feature( invert ),
+                                                              _abstract_container_color( abstract_container_color ) {
+
+}
+
+/**
+ * Feature_Abstract_Container_Color class copy constructor
+ */
+Feature_Abstract_Container_Color::
+Feature_Abstract_Container_Color( const Feature_Abstract_Container_Color& other ) : Feature( other ),
+                                                              _abstract_container_color( other._abstract_container_color ){
+
+}
+
+/**
+ * Feature_Abstract_Container_Color class destructor
+ */
+Feature_Abstract_Container_Color::
+~Feature_Abstract_Container_Color() {
+
+}
+
+/** 
+ * Feature_Abstract_Container_Color class assignment operator
+ */
+Feature_Abstract_Container_Color&
+Feature_Abstract_Container_Color::
+operator=( const Feature_Abstract_Container_Color& other ){
+  _invert = other._invert;
+  _abstract_container_color = other._abstract_container_color;
+  return (*this);
+}
+
+/**
+ * returns the value of the fature
+ */
+bool
+Feature_Abstract_Container_Color::
+value( const unsigned int& cv,
+        const h2sl::Grounding* grounding,
+        const vector< pair< const h2sl::Phrase*, vector< h2sl::Grounding* > > >& children,
+        const h2sl::Phrase* phrase,
+        const World* world ){
+  const Abstract_Container* abstract_container = dynamic_cast< const Abstract_Container* >( grounding );
+  if( abstract_container != NULL ){
+    if( abstract_container->color() == _abstract_container_color ){
```

Comment:
I think we want to return _invert if this is false; this would make it consistent with other features like feature_region_container_type. so:

else{
  return _invert;
}